rancid-tomatillos/src/App/App.js

```diff
@@ -0,0 +1,75 @@
+import React, {Component} from "react";
+import {
+    BrowserRouter as Router,
+    Switch,
+    Route,
+    Redirect,
+} from "react-router-dom";
+import Dashboard from "../Dashboard/Dashboard";
+import Header from "../Header/Header";
+import Login from "../Login/Login";
+import ShowPage from "../MoviePage/MoviePage";
+
+import "./App.css";
+
+class App extends Component {
+    constructor() {
+        super();
+        this.state = {
+            loggedIn: false,
+            movies: null,
+            error: "",
+        };
+    }
+    async componentDidMount() {
+        let promise = await fetch(
+            "https://rancid-tomatillos.herokuapp.com/api/v2/movies/"
+        );
+        if (promise.ok) {
+            let result = await promise.json();
+            this.setState({movies: result.movies});
+        } else {
+            this.setState({error: promise.status});
+        }
+    }
+    toggleLogin = () => {
+        this.setState({loggedIn: !this.state.loggedIn});
+    };
+
+    render() {
+        return (
+            <Router>
+                <Header
+                    isAuthed={this.toggleLogin}
+                    loggedIn={this.state.loggedIn}
+                />
+                <div className="page-container">
+                    <Switch>
+                        <Route path="/users">
+                            <ShowPage />
+                        </Route>
+                        <Route
+                            path="/login"
+                            render={(props) => (
+                                <Login {...props} isAuthed={this.toggleLogin} />
+                            )}
+                        />
+                        <Route
+                            exact
+                            path="/"
+                            render={() =>
+                                this.state.error > 400 ? (
+                                    <Redirect to="/error" />
+                                ) : (
+                                        <Dashboard allMovies={this.state.movies} />
+                                    )
```

Comment:
I'm pretty happy with this error handling, need to add some stuffs for /error

Reply:
Yeah! I think this was a good implementation of past knowledge of modifying `Route.render()`

Reply:
I added an issue to add onto /error route display